### PR TITLE
Update controller product detail

### PR DESCRIPTION
**Pull request by at-dungvan**

### CODE REVIEW

File: app/controllers/ProductController.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	v.Vars["product"] = data
+	v.Name = "product/index"
+	this.Render(w, v)
+}
+
+var GetProductController = &ProductController{Render: renderView}
+
+type dataProduct struct {
```
Comment (at-duyho) [inline app/controllers/ProductController.go]: move it to DTO folder.

File: app/controllers/UserController.go [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
 		Password: strings.Trim(r.FormValue("password"), " "),
 	}
 	err := user.HashPassword()
-	if err == nil {
+	fmt.Println(user)
+	if err != nil {
 		flash.SetFlash(w, flash.Flash{"Password cannot hash!", flash.FlashError})
 	}
 	err = user.Create()
+	fmt.Println(err)
```
Comment (at-duyho) [inline app/controllers/UserController.go]: move it

File: app/controllers/UserController.go [hunk 3]
```diff
@@ -1,25 +1,30 @@
...
 	user := models.User{
-		Name: strings.Trim(r.FormValue("name"), " "),
-		Email: strings.Trim(r.FormValue("email"), " "),
+		Name:     strings.Trim(r.FormValue("name"), " "),
+		Email:    strings.Trim(r.FormValue("email"), " "),
 		Password: strings.Trim(r.FormValue("password"), " "),
 	}
 	err := user.HashPassword()
-	if err == nil {
+	fmt.Println(user)
```
Comment (at-duyho) [inline app/controllers/UserController.go]: remove it

File: app/models/Product.go [hunk 4]
```diff
@@ -13,6 +15,42 @@ type Product struct {
...
+	}
+	defer stmt.Close()
+	err = stmt.QueryRow(this.Name, this.Description, this.Quantity, this.Price, time.Now(), time.Now()).Scan(&this.ID)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (this *Product) GetAll(limit int) (products []Product, err error) {
```
Comment (at-duyho) [inline app/models/Product.go]: get all but have limit?

File: app/controllers/Controller.go [hunk 5]
```diff
@@ -28,3 +28,15 @@ type LoginController struct {
 type UserController struct {
 	Render render
 }
+
+type CartController struct {
```
Comment (at-duyho) [inline app/controllers/Controller.go]: move it to CartController.go file

File: app/controllers/Controller.go [hunk 6]
```diff
@@ -28,3 +28,15 @@ type LoginController struct {
...
+	Render render
+}
+
+type ProductController struct {
```
Comment (at-duyho) [inline app/controllers/Controller.go]: move it to ProductController.go file

File: app/controllers/Controller.go [hunk 7]
```diff
@@ -28,3 +28,15 @@ type LoginController struct {
...
+	Render render
+}
+
+type CheckoutController struct {
```
Comment (at-duyho) [inline app/controllers/Controller.go]: move it to CheckoutController.go file

File: app/models/Product.go (hunk 4 above)
Comment (at-dungvan) [reply app/models/Product.go]: define the limit for pagination in future!

File: main.go [hunk 8]
```diff
@@ -4,6 +4,8 @@ import (
 	"encoding/json"
 	"os"
 
+	// "github.com/goweb3/app/models"
```
Comment (at-duyho) [inline main.go]: remove it

File: app/models/Product.go (hunk 4 above)
Comment (at-duyho) [reply app/models/Product.go]: so! you have to define another function.

**Pull request merged**